frontend/src/components/c-ramp.vue

```diff
@@ -111,29 +121,31 @@ export default {
       if (this.isDeletesContribution(slice)) {
         return this.deletesContributionRampSize;
       }
-      const newSize = 100 * (slice.insertions / this.avgsize);
+      // '+' unary operator here attempts to convert this.avgsize to number, if it is not already one
+      const newSize = 100 * (slice.insertions / +this.avgsize);
       return Math.max(newSize * this.rampSize, 0.5);
     },
-    getContributionMessage(slice, commit) {
-      let title = '';
-      if (this.tframe === 'commit') {
+    // commit of type CommitResult must be provided if tframe === 'commit', only pass null if tframe !== 'commit
+    getContributionMessage(slice: Commit, commit: CommitResult | null) {
+      if (this.tframe === 'commit' && commit !== null) {
         return `[${slice.date}] ${commit.messageTitle}: +${commit.insertions} -${commit.deletions} lines `;
       }
 
-      title = this.tframe === 'day'
-            ? `[${slice.date}] Daily `
-            : `[${slice.date} till ${slice.endDate}] Weekly `;
+      let title = this.tframe === 'day'
+          ? `[${slice.date}] Daily `
+          : `[${slice.date} till ${slice.endDate}] Weekly `;
       title += `contribution: +${slice.insertions} -${slice.deletions} lines`;
       return title;
     },
-    openTabZoom(user, slice, evt) {
+    // Only to be used when this.tframe !== 'Commit'
```

Comment:
If this shouldn't be called when `tframe` is `Commit`, would it be beneficial to explicitly check for this within the method and abort/exit early? Will it break the application if the method was called when this condition is met?

Reply:
After playing around with removing `openTabZoom` and using it with `this.tframe === 'Commit'`, it seems that `openTabZoom` in `c-ramp` is somehow redundant, at least on my Windows OS. It seems that the logic for opening the Zoom tab is in `c-summary-charts.vue`, under the `openTabZoom` method. As I am unsure of its significance, I will remove the comment and keep the method. Perhaps this can be investigated in a separate issue/PR.

Reply:
I can't test right now, but if i'm not wrong the method is called when the granularity is set to "day" or "week" (one of the drop-downs on the top left), and you click on a ramp on the summary charts panel - it should open the zoom view for commits represented by the ramp that you clicked on. This was on MacOS though.

Reply:
I've done further testing as you have described, and it seems that at least of Windows OS, the `openTabZoom` method might be redundant. Here are some screenshots to showcase the testing process.

Summary report settings
![image](https://github.com/reposense/RepoSense/assets/95712150/917adcde-af43-4fe9-bf2c-6d7bbfd4d0b3)

Prop data stored in CZoom CRamp before removal of `openTabZoom` method
![image](https://github.com/reposense/RepoSense/assets/95712150/1af1ec2e-7fbf-4a37-8397-d72133009e4b)

Prop data stored in CZoom CRamp after removal of `openTabZoom` method
![image](https://github.com/reposense/RepoSense/assets/95712150/990e9d5a-350f-4891-8d7f-a0a03671cfd0)

As seen from the prop data, the data seems to be identical before and after removing the `openTabZoom` method, indicating a possibly redundant `openTabZoom` method in `c-ramp.vue`. It seems that the `openTabZoomSubrange` method in `c-summary-charts.vue` that was provided as an event handler in the following code may be either overriding any data modified by the `openTabZoom` method in `c-ramp.vue`, or that the method is not even triggering. The following is an image of the code segment where the `openTabZoomSubrange` is passed as an event handler to the `summary-chart__ramp` element containing the `c-ramp` component that is being used to select a zoom range.

![image](https://github.com/reposense/RepoSense/assets/95712150/de4fd8be-8501-4815-8bb6-9d925ac10299)

Perhaps we could test in a similar manner for MacOS and Linux to ensure that the `openTabZoom` method in `c-ramp.vue` is truly redundant.

Notes:
* Testing done on Windows OS, Windows 11
* The data debugging tool used is the Chrome Vue extension recommended in the DG
* Page was refreshed and the same zoom region reselected before screenshots were taken

Reply:
Sorry for the delay! Here's a reproduction of what I was referring to in my earlier message - this calls the `openTabZoom` method in `c-ramp.vue`:


<details>

![ramp-zoom](https://github.com/reposense/RepoSense/assets/45852430/2d8d3b73-55d1-43bb-9463-ff261c8d7b4a)
</details>


<details>
<summary>Difference in `commits` object received by zoom tab</summary>

![ramp-zoom2](https://github.com/reposense/RepoSense/assets/45852430/d56df452-d75c-44be-9172-934665c92483)
</details>

Reply:
Thanks for the clarification, I now understand what you meant. I have modified the `openTabZoom` method such that the total insertions and deletions are calculated to allow for a safe creation of `Commit`.

Reply:
Sorry for the delay. Did some investigation regarding this issue, will summarize for the benefit of others/future contributors as well:

- In the current code, `DailyCommit[]` is passed in place of `Commit[]` to the zoom view, which expects `Commit[]`. `Commit` is essentially a supertype of `DailyCommit` - it contains [two additional fields, `insertions` and `deletions`](https://github.com/reposense/RepoSense/blob/master/frontend/src/types/types.ts#L30).
- This means that anytime code in the zoom view (provided the zoom view is generated by this method call, which happens when a user clicks on a commit in the ramp with granularity set to day/week as outlined above & in the video), accesses these properties (`commit.insertions`/`commit.deletions`), something might break
- There are currently 2 bugs that I found with the zoom view that is generated from this action,
  - Sorting by lines of code in the zoom view doesn't work properly, since it sorts by `commit.insertions`, which would be undefined (https://github.com/reposense/RepoSense/blob/master/frontend/src/views/c-zoom.vue#L191)
  - The message body of commits aren't shown, as can be seen in the video in the earlier comment. This might be a potential regression caused by #1866, as the `isOpen` property that enables the toggling of message bodies is only added to the commits in `user.commits` instead of `user.dailyCommits`. (https://github.com/reposense/RepoSense/pull/1866/files#diff-5419037f2fc8c426a6a54e0614ee8033d10e16426d30573486c27e5811036869)

The fix you introduced fixes the first bug, but not the second. I'm actually not sure what the purpose of the original line `zoomUser.commits = user.dailyCommits;` is. It seems that by removing the line (and leaving `zoomUser.commits` alone), both of the above bugs don't occur. 

Given that these bugs are due to existing code and that this fix technically changes logic but doesn't fully fix the existing bugs, it's my preference that we don't introduce the new adding of insertions/deletions in this PR. IMO we can just do the original unsafe typecast (`as Commit[]`) and leave a comment warning that this typecast is unsafe, and file the current bugs as an issue to be fixed in a separate PR, since fixing those are out of the scope of migrating the file to TypeScript. I think this is just one of the instances where converting files to TypeScript help us uncover existing bugs.

The above approach is just my preference - please let me know if you have any thoughts on it (other reviewers please feel free to chime in!)

Reply:
Once again, thanks for investigating the above issue. Regarding the message body bug, I have come up with a modification to my fix that would address the message body bug, as well as the sorting bug, however, I do agree with the idea of creating a separate issue and PR to address this, for the purposes of maintaining a clear scope for each PR.

Reply:
Sounds good! Will approve this PR once the fix is reverted and feel free to request a review when you open the new PR. Thanks!